File: CloudFlow/src/ro/cloudSoft/cloudDoc/presentation/client/shared/model/calendar/BirthdayCalendarEventModel.java
package ro.cloudSoft.cloudDoc.presentation.client.shared.model.calendar;

import java.util.Date;

import ro.cloudSoft.cloudDoc.domain.calendar.CalendarRepeatEventEnum;

public class BirthdayCalendarEventModel extends CalendarEventModel {
	
	public BirthdayCalendarEventModel() {
		super(CalendarEventType.BIRTHDAY);
	}
	
	private CalendarRepeatEventEnum repeat;
	private Date birthdate;
	private Date birthdateEvent;
	
	public CalendarRepeatEventEnum getRepeat() {
		return repeat;
	}
	public void setRepeat(CalendarRepeatEventEnum repeat) {
		this.repeat = repeat;
	}
	public Date getBirthdate() {
		return birthdate;
	}
	public void setBirthdate(Date birthdate) {
		this.birthdate = birthdate;
	}
	public Date getBirthdateEvent() {
		return birthdateEvent;
	}
	public void setBirthdateEvent(Date birthdateEvent) {
		this.birthdateEvent = birthdateEvent;
	}
}

File: CloudFlow/FrontEnd-App/src/app/shared/model/calendar/birthday-calendar-event.model.ts
import { JsonObject, JsonProperty } from "json2typescript";
import { OrganizationEntityModel } from "../organization-entity.model";
import { JsonDateConverter } from "../../json-mapper";
import { CalendarEventModel } from "./calendar-event.model";

@JsonObject
export class BirthdayCalendarEventModel extends CalendarEventModel {

	@JsonProperty("repeat", String)
	public repeat: string = null;
	
	@JsonProperty("birthdate", JsonDateConverter)
	public birthdate: Date = null;

	@JsonProperty("birthdateEvent", JsonDateConverter)
	public birthdateEvent: Date = null;
}